```
======================================================================
ERROR: test_vary_on_func (corehq.util.tests.test_quickcache.QuickcacheTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/tests/test_quickcache.py", line 182, in test_vary_on_func
    self.assertRegex(key, 'quickcache.cached_fn.[a-z0-9]{8}/sa1')
AttributeError: 'QuickcacheTest' object has no attribute 'assertRegex'
----------------------------------------------------------------------
```

--- a/corehq/apps/accounting/utils.py
+++ b/corehq/apps/accounting/utils.py
@@ -99,7 +99,14 @@ def get_change_status(from_plan_version, to_plan_version):
     return adjustment_reason, downgraded_privs, upgraded_privs
 
 
-@quickcache(['domain', 'privilege_slug'], timeout=10)
+def domain_has_privilege_cache_args(domain, privilege_slug, **assignment):
+    return [
+        domain.name if isinstance(domain, Domain) else domain,
+        privilege_slug
+    ]
+
+
+@quickcache(domain_has_privilege_cache_args, timeout=10)
 def domain_has_privilege(domain, privilege_slug, **assignment):
     from corehq.apps.accounting.models import Subscription
     try:

--- a/corehq/util/tests/test_quickcache.py
+++ b/corehq/util/tests/test_quickcache.py
@@ -179,4 +179,4 @@ class QuickcacheTest(SimpleTestCase):
             pass
 
         key = cached_fn.get_cache_key({'name': 'a1'})
-        self.assertRegex(key, 'quickcache.cached_fn.[a-z0-9]{8}/sa1')
+        self.assertRegexpMatches(key, 'quickcache.cached_fn.[a-z0-9]{8}/sa1')

